mantidproject/mantid
# Fix crash reported in ALC interface with "Auto"

Fix a crash reported in the ALC interface

The interface froze when used with the "Auto" checkbox ticked in a network drive with lots of instrument data in it. The "Auto" function had been tested on directories of .nxs files, but this contained many different file types and other directories - the non-NeXus files weren't being ignored correctly, causing an infinite loop when trying to find the most recently modified file. 

Fixed the bug and moved the function into its own class so that it could be tested.
Added a test to prevent regression.

**To test:**

A new unit test is intended to cover the "finding most recent file in the directory" functionality.
_Functional testing:_
Create a directory with some muon data files and also other (non-NeXus) files in it. 
(The crash was seen in `\\hifi\data` which contains nxs, raw, txt, log...)
Launch the ALC interface (Interfaces/Muon/ALC) and select one of the muon files as the first run. Check "Auto" by the last run box and try to load the data.
- Interface should not crash or freeze
- Should pick up new files added/removed automatically
- Should deal with the case where the newest file is not a NeXus file
- Try generating a crash with bad input

Fixes #15662.

The [release notes](https://github.com/mantidproject/mantid/blob/fc485521e39f96cdbd13a23962da31a636b9fc5b/docs/source/release/v3.7.0/muon.rst) have been updated.

**To do:** update James when fix is in place

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
